frontend/components/auction/debt/DebtAuctionTransactionFlow.vue

```diff
@@ -0,0 +1,158 @@
+        />
+        <div class="mb-4">
+            <WalletConnectionCheckPanel
+                :wallet-address="walletAddress"
+                :is-loading="isConnectingWallet"
+                :is-explanations-shown="isExplanationsShown"
+                :is-correct.sync="isWalletConnected"
+                @connectWallet="$emit('connectWallet')"
+                @disconnectWallet="$emit('disconnectWallet')"
+            />
+            <HighestDebtBidCheckPanel
+                :auction="auction"
+                :wallet-address="walletAddress"
+                :disabled="!isWalletMKRCheckPassed || !isAllowanceAmountCheckPassed || !isActive"
+                :is-loading="auctionActionState === 'bidding'"
+                :bid-amount="inputBidAmount || auction.nextMaximumLotReceived"
+                :is-explanations-shown="isExplanationsShown"
+                :is-correct.sync="isHighestBidder"
+                @bid="$emit('bid', $event)"
+            />
+            <CollectAuctionPanel
+                :auction="auction"
+                :wallet-address="walletAddress"
+                :is-collecting="auctionActionState === 'collecting'"
+                currency="MKR"
+                @collect="$emit('collect')"
+            />
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import type { DebtAuction, SurplusAuctionStates, CompensationAuctionActionStates } from 'auctions-core/src/types';
+import Vue from 'vue';
+import { Alert } from 'ant-design-vue';
+import BigNumber from 'bignumber.js';
+import HighestDebtBidCheckPanel from '../../panels/HighestDebtBidCheckPanel.vue';
+import DebtAuctionBidTransactionTable from './DebtAuctionBidTransactionTable.vue';
+import WalletConnectionCheckPanel from '~/components/panels/WalletConnectionCheckPanel.vue';
+import TextBlock from '~/components/common/other/TextBlock.vue';
+import CollectAuctionPanel from '~/components/panels/CollectAuctionPanel.vue';
+
+export default Vue.extend({
+    components: {
+        CollectAuctionPanel,
+        HighestDebtBidCheckPanel,
+        DebtAuctionBidTransactionTable,
+        TextBlock,
+        Alert,
+        WalletConnectionCheckPanel,
+    },
+    props: {
+        auction: {
+            type: Object as Vue.PropType<DebtAuction>,
+            required: true,
+        },
+        auctionActionState: {
+            type: String as Vue.PropType<CompensationAuctionActionStates>,
+            default: null,
+        },
+        walletAddress: {
+            type: String,
+            default: null,
+        },
+        walletDai: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: undefined,
+        },
+        walletVatDai: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: undefined,
+        },
+        allowanceDai: {
+            type: Object as Vue.PropType<BigNumber>,
+            default: undefined,
+        },
+        tokenAddress: {
```

Comment:
Which taken is it? Can you please name it?

Reply:
This token is also present in the SurplusFlow. It is displayed in the VAT check panel.

See mock below:
![Screenshot 2022-08-10 at 09 53 40](https://user-images.githubusercontent.com/30908158/183846307-1b9388c0-62ce-439b-9823-60f6d6a3a2c5.png)

Reply:
Is it DAI or MKR or some other token? It's just not clear from the prop name. Since this panel doesn't accept `currency`

Reply:
The panel does accept Currency as a prop:

```html
<WalletVATCheckPanel
  :is-correct.sync="isEnoughInWallet"
  :wallet-amount="walletAmount"
  :wallet-vat-amount="walletVatAmount"
  :desired-amount="desiredAmount"
  :network="network"
  :token-address="tokenAddress"
  :currency="currency"
  :disabled="disabled"
  :is-loading="isLoading"
  :is-explanations-shown="isExplanationsShown"
  @refresh="$emit('refresh')"
/>
```

I did not want to have separate props for a MKR and DAI token, as that would require more unneeded props. The MKR Token is fetched [here](https://github.com/sidestream-tech/unified-auctions-ui/blob/main/frontend/store/surplus.ts#L283). I assume we can easily modify the code to get the DAI token for this page.